pkg/scalers/kafka_scaler.go

```diff
@@ -295,11 +314,13 @@ func (s *kafkaScaler) getLagForPartition(partition int32, offsets *sarama.Offset
 	}
 
 	var lag int64
-	// For now, assume a consumer group that has no committed
-	// offset will read all messages from the topic. This may be
-	// something we want to allow users to configure.
+
 	if consumerOffset == sarama.OffsetNewest || consumerOffset == sarama.OffsetOldest {
-		lag = latestOffset
+		if s.metadata.consumerOffsetReset == latest {
+			lag = 0
```

Comment:
shouldn't this be `lag = latestOffset` ?

Reply:
when you have no offset committed and you create a new consumer with reset latest policy then the lag should be 0 since the consumer is aligned with the latest offset on the topic.

Reply:
ok, I am not a Kafka expert, but the properties naming is pretty confusing